_source/_posts/2022-04-06-react-azure-functions.md

```diff
@@ -0,0 +1,415 @@
+                Login
+              </button>
+            </p>
+          </div>
+        )}
+        {authState?.isAuthenticated && (
+          <div>
+            <p className="lead">To Create your badge, upload your image below</p>
+            <input onChange={acceptImage} name="image" type="file" />
+            <button className="btn btn-primary" onClick={createBadge}>
+              Upload
+            </button>
+            <br />
+            {display && <img className="pt-4" alt="your MIB badge" src={display}></img>}
+          </div>
+        )}
+
+        <footer
+          className="bg-light text-center fixed-bottom"
+          style={{
+            width: "100%",
+            padding: "0 15px",
+          }}
+        >
+          <p>
+            A Small demo using <a href="https://developer.okta.com/">Okta</a> to Secure an{" "}
+            <a href="https://azure.microsoft.com/en-us/services/app-service/static/">Azure Static Web App </a> with a serverless{" "}
+            <a href="https://azure.microsoft.com/en-us/services/functions/">Function</a>
+          </p>
+          <p>
+            By <a href="https://github.com/nickolasfisher">Nik Fisher</a>
+          </p>
+        </footer>
+      </main>
+    </div>
+  );
+}
+
+export default Home;
+```
+{% endraw %}
+
+This file contains the bulk of your logic. First, it provides Login/Logout functionality using the `useOktaAuth` hook. With this hook, you can determine the user's authenticated state. If the user is not, then prompt them to do so; otherwise, you will allow them to use the badge creator.
+
+The badge creator logic prompts users to upload a photo of themselves for the template. It then posts this to the nebulous `api/CreateBadge`. This route stands for the CreateBadge function that you will create later in this article. Azure will know how to find that route whether you're running this application locally on Azure's emulator or Azure's infrastructure. It will even be able to route to the appropriate environment on Azure's servers.
+
+A note here, you would likely expect to send the `accessToken` in the `Authorization` header; however, Azure overwrites the `Authorization` header with its token by default. You can eliminate this step on the standard pricing model by using the custom providers in the Static Web App and the Function. However, you will need to use this workaround on the free model.
+
+In this tutorial, the client sends the username from the ID Token. Ideally, the Azure Function should retrieve the username from making a call to the `/userprofile` endpoint. By having the Azure Function handle this, you can ensure you get the accurate username and not rely on the client potentially sending something inaccurate.
+
+One other note is that environment variables do not work at this time on Static Web Apps. If you attempt to use `process.env.{variable}` in your code and set it in the application settings, it will not work.
```

Comment:
```suggestion
One other note: Environment variables do not work at this time on Static Web Apps. If you attempt to use `process.env.{variable}` in your code and set it in the application settings, it will not work.
```
conciseness